### ISSUE TITLE
Extraction error: Only if the required item is first seen in the chest it is taken by the system.

### ISSUE THREAD

**Issue by KTHyperion** [RS485/LogisticsPipes#1705]
LogisticsPipes version: logisticspipes-0.10.4.49.jar

**CONTEXT**:
The machine has ItemSink, Provider, Adv. Extractor modules;
ItemSink Module was configured to require some items from the system.
The chest (let it be vanilla or IronChest) has Polymorphic ItemSink, Provider, Adv. Extractor;
This chest has the items required by the machine's ItemSink.

**VALIDATION TEST:**
When a machine has an ItemSink module requiring some items,
And the item is in a chest who is connected to the system via Provider Module,
So you expect the item to be removed from the chest

**TEST FAIL:**
The item is only removed by the chest IF and ONLY if the item is the first item found in the chest when the iteration starts.

**EVIDENCE PHOTOS:**

The items who are requested by the system:
![ItemsinkRequest](https://github.com/RS485/LogisticsPipes/assets/135667778/37193eaf-e52c-4ba6-814f-4139a06f34cd)

The modules in the chest
![ChestModules](https://github.com/RS485/LogisticsPipes/assets/135667778/8774f32b-c469-4f1b-b37b-eccc56f69d3c)

The provider configuration for the chest:
![ProviderModule](https://github.com/RS485/LogisticsPipes/assets/135667778/c17f4c24-0fcf-4eea-8f3d-772fd29abdbd)

The extractor configuration for the chest
![ExtractorModule](https://github.com/RS485/LogisticsPipes/assets/135667778/752311d7-84c8-4570-b425-7475672721ea)

**EVIDENCE VIDEOS:**
(The video resolutions needed to be lowered since GitHub limits the video size)

IronChest Mod Manual Testing
https://github.com/RS485/LogisticsPipes/assets/135667778/f9497c6e-a6a3-4516-8be4-bd63ac6eeaba

Vanilla Minecraft Chest Manual Testing
https://github.com/RS485/LogisticsPipes/assets/135667778/436d6b78-e936-42ce-90f6-82bd350b946c

ADDITIONAL COMMENTS:
Despite the videos not showing the items not flowing by a prolonged time, you can make the test by yourself: if the item required is NOT the first item seen by the system iteration, it will not be (ever) extracted,

**Comment by bziemons** [comment]
Is this reproducible with a standard vanilla chest?

**Comment by KTHyperion** [comment]
Yes, it is. The second video published in the end of the report shows the same event happening with Vanilla Chest.
I'm going to record a new, longer video and add to a new comment.

**Comment by KTHyperion** [comment]
https://github.com/RS485/LogisticsPipes/assets/135667778/fda7ab07-ce23-4058-98b4-b8c2667e8139

**Comment by bziemons** [comment]
> Yes, it is. The second video published in the end of the report shows the same event happening with Vanilla Chest. I'm going to record a new, longer video and add to a new comment.

Sorry, my bad :facepalm: Thank you for your detailed bug report!

**Comment by KTHyperion** [comment]
NEW UPDATE: I've been making some tests with and without other mods. The bug described above still shows itself when only Logistics Pipes (and Extreme Reactors for energy source purposes) is installed, BUT when an item is required by Active Supplier Module, this bug doesn't happen, it acts as normal. The problem happens only when required by an Itemsink Module.
